keygen/src/keygen.rs

```diff
@@ -457,64 +458,64 @@ fn main() -> Result<(), Box<dyn error::Error>> {
                 )
         )
         .subcommand(
-            SubCommand::with_name("pubkey")
+            Command::new("pubkey")
                 .about("Display the pubkey from a keypair file")
-                .setting(AppSettings::DisableVersion)
+                .disable_version_flag(true)
                 .arg(
-                    Arg::with_name("keypair")
+                    Arg::new("keypair")
                         .index(1)
                         .value_name("KEYPAIR")
                         .takes_value(true)
                         .help("Filepath or URL to a keypair"),
                 )
                 .arg(
-                    Arg::with_name(SKIP_SEED_PHRASE_VALIDATION_ARG.name)
+                    Arg::new(SKIP_SEED_PHRASE_VALIDATION_ARG.name)
                         .long(SKIP_SEED_PHRASE_VALIDATION_ARG.long)
                         .help(SKIP_SEED_PHRASE_VALIDATION_ARG.help),
                 )
                 .arg(
-                    Arg::with_name("outfile")
-                        .short("o")
+                    Arg::new("outfile")
+                        .short('o')
                         .long("outfile")
                         .value_name("FILEPATH")
                         .takes_value(true)
                         .help("Path to generated file"),
                 )
                 .arg(
-                    Arg::with_name("force")
-                        .short("f")
+                    Arg::new("force")
+                        .short('f')
                         .long("force")
                         .help("Overwrite the output file if it exists"),
                 )
         )
         .subcommand(
-            SubCommand::with_name("recover")
+            Command::new("recover")
                 .about("Recover keypair from seed phrase and optional BIP39 passphrase")
-                .setting(AppSettings::DisableVersion)
+                .disable_version_flag(true)
                 .arg(
-                    Arg::with_name("prompt_signer")
+                    Arg::new("prompt_signer")
                         .index(1)
                         .value_name("KEYPAIR")
                         .takes_value(true)
-                        .validator(is_prompt_signer_source)
+                        .validator(|s| is_prompt_signer_source(s))
```

Comment:
superfluous closure?

Reply:
without this closure it doesn't compile with error:
```
error: implementation of `FnOnce` is not general enough
   --> keygen/src/keygen.rs:500:26
    |
500 |                         .validator(is_prompt_signer_source)
    |                          ^^^^^^^^^ implementation of `FnOnce` is not general enough
    |
    = note: `fn(&'2 str) -> Result<(), std::string::String> {is_prompt_signer_source::<&'2 str>}` must implement `FnOnce<(&'1 str,)>`, for any lifetime `'1`...
    = note: ...but it actually implements `FnOnce<(&'2 str,)>`, for some specific lifetime `'2`

error: could not compile `solana-keygen` due to previous error
```
I found explanation [here](https://users.rust-lang.org/t/implementation-of-fnonce-is-not-general-enough/68294/3).

Reply:
even when adapting the signature of `is_prompt_signer_source` to explicitly take `&str`? I'd think it's just not able to resolve `AsRef<str>` to `&str` automatically :thinking:

Reply:
If I use `&str` it works (updated). Looks like `validator` is always assuming `&str` so might be unnecessary to have generics there. My only concern is that all the functions in the `input_validators.rs` are using the same generic signature and may be there is a reason behind which is unknown to me. So I got rid of generic only in `is_prompt_signer_source`, others I left as they are to modify in next PRs by demand, if makes sense.

Reply:
cool! agreed updating all of the validators in a future PR is the right thing